webcomponents/tool/src/main/frontend/packages/sakai-date-picker/src/SakaiDatePicker.js

```diff
@@ -43,7 +43,9 @@ export class SakaiDatePicker extends SakaiElement {
 
     if (value) {
       this._epochMillis = value;
-      this.isoDate = (new Date(value + parseInt(getOffsetFromServerMillis()))).toISOString().substring(0, 16);
+      const instant = Temporal.Instant.fromEpochMilliseconds(value);
+      const zonedDateTime = instant.toZonedDateTimeISO({ timeZone: getTimezone() });
```

Comment:
has anyone tested this when a user sets their tz via preferences?

Reply:
yes i tested today.... previous code relied on getOffsetFromServerMillis which is an offset for ***today***. obviously more sane to just let a library do the manipulation using the timezone. 

I'm not sure why the webcomponent code converts dates to millis at all. maybe @adrianfish can chime in.

Reply:
Not sure at this point. Maybe because of way we handle stuff on the server? Nice change, by the way.